tests/recovery.py

```diff
@@ -254,8 +259,27 @@ def test_recover_service_with_wrong_identity(network, args):
         snapshots_dir=snapshots_dir,
     )
 
-    recovered_network.recover(args)
+    # Must fail with a dedicated error message if requesting a receipt for a TX
+    # from past epochs, as soon as ledger secrets are not yet available,
+    # therefore no proof can't be generated.
+    primary, _ = recovered_network.find_primary()
+    with primary.client() as cli:
+        curr_view, curr_seq = (
+            cli.get("/node/commit").body.json()["transaction_id"].split(".")
+        )
+        response = cli.get(f"/node/receipt?transaction_id={last_view}.{last_seq}")
+        assert response.status_code == http.HTTPStatus.NOT_FOUND, response
+        assert (
+            "not signed by the current servic"
+            in response.body.json()["error"]["message"]
+        ), response
 
+    # TX from the current epoch though can be verified, as soon as the caller
+    # trusts the current service identity.
+    receipt = primary.get_receipt(curr_view, curr_seq).json()
+    verify_receipt(receipt, recovered_network.cert, is_signature_tx=True)
```

Comment:
This is not ideal, because it relies on the current transaction being a signature. That's probably true most of the time, but maybe not if a JWT refresh was recently issued for example. It would be preferable to walk back from current to the recovery txid, and make sure we can verify actual receipts.

Reply:
Done for multiple Txs, but not for all
Reasons
- didn't find out how to easily iterate over all ledger files
- can consume quite some time, probably we can do smth like that in a long-test run